python/pyiceberg/catalog/glue.py

```diff
@@ -0,0 +1,549 @@
+            parameters[k] = v
+    database_input[PROP_GLUE_DATABASE_PARAMETERS] = parameters
+    return database_input
+
+
+def _write_metadata(metadata: TableMetadata, io: FileIO, metadate_path: str) -> None:
+    ToOutputFile.table_metadata(metadata, io.new_output(metadate_path))
+
+
+class GlueCatalog(Catalog):
+    @staticmethod
+    def identifier_to_database(
+        identifier: Union[str, Identifier], err: Union[Type[ValueError], Type[NoSuchNamespaceError]] = ValueError
+    ) -> str:
+        tuple_identifier = Catalog.identifier_to_tuple(identifier)
+        if len(tuple_identifier) != 1:
+            raise err(f"Invalid database, hierarchical namespaces are not supported: {identifier}")
+
+        return tuple_identifier[0]
+
+    @staticmethod
+    def identifier_to_database_and_table(
+        identifier: Union[str, Identifier],
+        err: Union[Type[ValueError], Type[NoSuchTableError], Type[NoSuchNamespaceError]] = ValueError,
+    ) -> Tuple[str, str]:
+        tuple_identifier = Catalog.identifier_to_tuple(identifier)
+        if len(tuple_identifier) != 2:
+            raise err(f"Invalid path, hierarchical namespaces are not supported: {identifier}")
+
+        return tuple_identifier[0], tuple_identifier[1]
+
+    def __init__(self, name: str, **properties: str):
+        super().__init__(name, **properties)
+        self.glue = boto3.client(GLUE_CLIENT)
+
+    def _convert_glue_to_iceberg(self, glue_table: Dict[str, Any]) -> Table:
+        properties: Properties = glue_table.get(PROP_GLUE_TABLE_PARAMETERS, {})
+
+        if TABLE_TYPE not in properties:
+            raise NoSuchPropertyException(
+                f"Property {TABLE_TYPE} missing, could not determine type: "
+                f"{glue_table[PROP_GLUE_TABLE_DATABASE_NAME]}.{glue_table[PROP_GLUE_TABLE_NAME]}"
+            )
+        glue_table_type = properties[TABLE_TYPE]
+
+        if glue_table_type.lower() != ICEBERG:
+            raise NoSuchIcebergTableError(
+                f"Property table_type is {glue_table_type}, expected {ICEBERG}: "
+                f"{glue_table[PROP_GLUE_TABLE_DATABASE_NAME]}.{glue_table[PROP_GLUE_TABLE_NAME]}"
+            )
+
+        if METADATA_LOCATION not in properties:
+            raise NoSuchPropertyException(
+                f"Table property {METADATA_LOCATION} is missing, cannot find metadata for: "
+                f"{glue_table[PROP_GLUE_TABLE_DATABASE_NAME]}.{glue_table[PROP_GLUE_TABLE_NAME]}"
+            )
+        metadata_location = properties[METADATA_LOCATION]
+
+        io = load_file_io(properties=self.properties, location=metadata_location)
+        file = io.new_input(metadata_location)
+        metadata = FromInputFile.table_metadata(file)
+        return Table(
+            identifier=(glue_table[PROP_GLUE_TABLE_DATABASE_NAME], glue_table[PROP_GLUE_TABLE_NAME]),
+            metadata=metadata,
+            metadata_location=metadata_location,
+            io=self._load_file_io(metadata.properties),
```

Comment:
I've rebased and updated the PR. This is the main change I made after the rebase.